config/routes.rb

```diff
@@ -3,5 +3,9 @@
 
   devise_for :users
 
-  resources :events, only: %i[new create update destroy]
+  resources :events, only: %i[new create update destroy] do
+    resources :bookings, only: %i[create update] do
+      patch :cancelation, on: :member
```

Comment:
I understand that the :cancelation is a custom route. But what is the "on: :member" for?

Reply:
When you create custom routes for a controller they are either members or collection
Usually you see it written as
```
resources :bookings do
  member do
    post "cancelation"
  end
end
```
instead

member means that it is a route that requires an id
collection means that it is a route that does not require an id

In this case because we want to cancel a specific id that means we need to pass the booking's id inside the route so it needs to be a member route

Reply:
got it thanks.